core/src/main/java/org/elasticsearch/search/internal/DefaultSearchContext.java

```diff
@@ -127,7 +128,9 @@
     public DefaultSearchContext(long id, ShardSearchRequest request, SearchShardTarget shardTarget,
                          Engine.Searcher engineSearcher, IndexService indexService, IndexShard indexShard,
                          ScriptService scriptService, PageCacheRecycler pageCacheRecycler,
-                         BigArrays bigArrays, Counter timeEstimateCounter, ParseFieldMatcher parseFieldMatcher) {
+                         BigArrays bigArrays, Counter timeEstimateCounter, ParseFieldMatcher parseFieldMatcher,
+                         TimeValue timeout
```

Comment:
I think that the name `timeout` for this parameter is appropriate. It is used to set the `timeout` for the context unless it is later overwritten using the setter. While the current uses set this parameter from the global default search timeout setting, the definition does not preclude the possibility of it being set from other sources.